pac4j-core/src/main/java/org/pac4j/core/engine/DefaultSecurityLogic.java

```diff
@@ -150,7 +150,7 @@ public R perform(final C context, final Config config, final SecurityGrantedAcce
             } else {
 
                 logger.debug("no matching for this request -> grant access");
-                return securityGrantedAccessAdapter.adapt(context, parameters);
+                return securityGrantedAccessAdapter.adapt(context, Arrays.asList(), parameters);
```

Comment:
Thinking about this once more, I think we should load the profiles in that case too and pass them to the `securityGrantedAccessAdapter`...

Reply:
In that case, I'm not clear how I should load them... from session or not?

Also, in this branch of the if, we are in the case where the matcher does NOT apply, why would we thus load the profiles? I think we shouldn't and consider that pac4j should not interfere at all in that case with the request.

Reply:
You should load them using the `loadProfilesFromSession` method. In fact, you just need to move the first lines of source code before the `if (matchingChecker.matches(context, matchers, config.getMatchers())) {`.

I had the same reasoning that you have. Then, I wondered what would happen if I use the `ProfileManager` even if the `DefaultSecurityLogic` has not applied? The profiles would be loaded if they existed. This is why I came to this conclusion.

Reply:
But then maybe the question is opposite: why would we want to load profiles even with the `ProfileManager` if pac4j is not meant to have played into the processing of the request?
Currently none of the security logic behaviour is applied when the filter matcher does not apply, why would we want to depart from that for the profiles?
To me, if the filter's security logic's matcher does not apply, it should behave exactly the same as if I haven't had set a filter at all.
Actually, the marchers are only there because it's not possible to conditionally apply filters with most of the frameworks pac4j is integrated with, if it was possible, then there would be no matchers in pac4j's security logic IMHO.

So I don't agree with your conclusion, but if you think it's still better to load the profile, then no problem, it's yours to make the choice, I just want to be clear about the fact I think it is counter-intuitive to do so :)

Reply:
OK. It makes sense. I'm not always right, far from it ;-) I will merge it as is.

Reply:
hehe ;) thanks!